tests/components/mqtt/test_trigger.py

```diff
@@ -229,8 +231,9 @@ async def test_encoding_default(hass, calls, mqtt_mock):
     mqtt_mock.async_subscribe.assert_called_once_with("test-topic", ANY, 0, "utf-8")
 
 
-async def test_encoding_custom(hass, calls, mqtt_mock):
+async def test_encoding_custom(hass, calls, mqtt_mock_entry):
```

Comment:
There's an automatically used mock `setup_comp`, we should not need the `mqtt_mock_entry` here?